test-utils/src/commonMain/kotlin/FlowableTest.kt

```diff
@@ -36,52 +39,51 @@ import kotlinx.coroutines.withTimeout
 
 typealias TestBlock<T> = suspend(T) -> Unit
 typealias ActionBlock = suspend() -> Unit
+typealias FlowTestBlock<T> =  suspend FlowTest<T>.() -> Unit
 
 abstract class FlowableTest<T> : BaseTest() {
 
-    @BeforeTest
-    override fun beforeTest() {
-        super.beforeTest()
-
-        flowable = CompletableDeferred()
-        flowable.invokeOnCompletion { flowTest = FlowTest(flowable.getCompleted()) }
+    fun testWithFlow(block: FlowTestBlock<T>) = runBlocking {
+        FlowTest(this, ::flowable).testWithFlow(block)
     }
 
-    lateinit var flowable: CompletableDeferred<Flowable<T>>
-    lateinit var flowTest: FlowTest<T>
-
-    var filter: suspend(T) -> Boolean
-        get() = flowTest.filter
-        set(newValue) { flowTest.filter = newValue }
+    abstract fun flowable():Flowable<T>
 
-    fun testWithFlow(block: suspend FlowTest<T>.() -> Unit) {
-        flowTest.testWithFlow(block)
-    }
 }
 
-open class FlowTest<T>(private val flowable: Flowable<T>, private val coroutineScope: CoroutineScope = MultiplatformMainScope()) {
+open class FlowTest<T>(scope: CoroutineScope = MainScope(), val flowable:()->Flowable<T>):CoroutineScope by scope {
 
     open var filter: suspend(T) -> Boolean = { true }
 
     private val tests: MutableList<EmptyCompletableDeferred> = mutableListOf()
 
-    lateinit var job: Job
+    var job: Job? = null
 
     private lateinit var testChannel: Channel<Pair<TestBlock<T>, EmptyCompletableDeferred>>
 
-    private suspend fun endFlow() {
-        awaitTestBlocks() // get the final test blocks that were executed and check for exceptions
-        debug("test channel closed")
-        job.cancel()
-        debug("Ending flow")
+    suspend fun resetFlow() {
+        awaitTestBlocks()// get the final test blocks that were executed and check for exceptions
+        job?.cancel()
+        debug("Ending flow, job canceled")
+
+        firstTestBlock = true
         testChannel.close()
+
+        debug("test channel closed")
+
         tests.clear()
+        testChannel = Channel(Channel.UNLIMITED)
     }
 
-    protected val waitForTestToSucceed = 6000L * 10
+    protected val waitForTestToSucceed = 6000L * 10 / 10
+
+    private suspend fun awaitTestBlocks() {
+        if (tests.size == 0) {
+            println("await all test blocks, but none found")
+            return
+        }
 
-    suspend fun awaitTestBlocks() {
-        withTimeout(waitForTestToSucceed) { // only wait for one minute
+        withTimeout(waitForTestToSucceed) {// only wait for one minute
```

Comment:
Is not one minute

Reply:
3d14ef6b03c416de531fdf4686a32ba061719f87